felix/bpf/bpf.go

```diff
@@ -396,12 +396,32 @@ type perCpuMapEntry []struct {
 }
 
 type ProgInfo struct {
-	Name   string `json:"name"`
-	Id     int    `json:"id"`
-	Type   string `json:"type"`
-	Tag    string `json:"tag"`
-	MapIds []int  `json:"map_ids"`
-	Err    string `json:"error"`
+	Name   string      `json:"name"`
+	Id     int         `json:"id"`
+	Type   BPFProgType `json:"type"`
+	Tag    string      `json:"tag"`
+	MapIds []int       `json:"map_ids"`
+	Err    string      `json:"error"`
+}
+
+// BPFProgType is usually a string, but if the type is not known to bpftool, it
+// would be represented by an int. We do not care about those, but we must not
+// fail on parsing them.
+type BPFProgType string
+
+func (t *BPFProgType) UnmarshalJSON(data []byte) error {
+	if string(data) == "null" || string(data) == `""` || len(data) < 1 {
+		return nil
+	}
+
+	if data[0] == '"' {
+		s := string(data)[1 : len(data)-1]
```

Comment:
Probably OK here but this won't handle escape codes and the like.  I'd be temped to hand off to the main JSON decoder just so I don't have to worry.

Reply:
could do that, although we know what the input is,don't we? We only parse bpftool.